specification/datacollaboration/resource-manager/Microsoft.DataCollaboration/preview/2020-05-04-preview/DataCollaboration.json

```diff
@@ -0,0 +1,5736 @@
+          "uniqueItems": false,
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "readOnly": true
+        },
+        "visibility": {
+          "description": "The referenced script visibility modifier.",
+          "enum": [
+            "ShowAll",
+            "HiddenContent"
+          ],
+          "type": "string",
+          "readOnly": true,
+          "x-ms-enum": {
+            "name": "visibility",
+            "modelAsString": true
+          }
+        }
+      }
+    },
+    "StorageAccountDataSet": {
+      "description": "An Azure storage account data set.",
+      "required": [
+        "properties",
+        "kind"
+      ],
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "#/definitions/DataSet"
+        }
+      ],
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/StorageAccountProperties",
+          "description": "Blob data source properties.",
+          "x-ms-client-flatten": true
+        }
+      },
+      "x-ms-discriminator-value": "StorageAccount"
+    },
+    "StorageAccountPath": {
+      "description": "Defines a single storage account path.",
+      "required": [
+        "containerName"
+      ],
+      "type": "object",
+      "properties": {
+        "consumerPath": {
+          "description": "Gets or sets the path on the consumer side where the dataset is to be mapped.",
+          "type": "string"
+        },
+        "containerName": {
+          "description": "Gets or sets the container name to share.",
+          "type": "string"
+        },
+        "providerPath": {
+          "description": "Gets or sets the path to file/folder within the container.",
+          "type": "string"
+        }
+      }
+    },
+    "StorageAccountProperties": {
+      "description": "Properties of the Storage account data set.",
+      "required": [
+        "storageAccountResourceId",
+        "subscriptionId",
+        "resourceGroup",
+        "storageAccountName"
+      ],
+      "type": "object",
+      "properties": {
+        "dataSetId": {
+          "description": "Unique id for identifying a data set resource",
+          "type": "string",
+          "readOnly": true
+        },
+        "location": {
+          "description": "Location of the storage account.",
+          "type": "string"
+        },
+        "paths": {
+          "description": "A list of paths under a given storage account.",
+          "uniqueItems": false,
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/StorageAccountPath"
+          }
+        },
+        "resourceGroup": {
+          "description": "Resource group of storage account",
+          "type": "string"
+        },
+        "storageAccountName": {
+          "description": "Storage account name of the source data set",
+          "type": "string"
+        },
+        "storageAccountResourceId": {
```

Comment:
The user should only need to input this and paths.  Everything else is either in the ID or can be looked up by your RP.